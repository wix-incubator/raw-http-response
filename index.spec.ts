import buildHttpResponse from "./index";

test("builds an empty http response", () => {
  const response = buildHttpResponse({
    status: 200,
    headers: { "X-My-Header": "Hello world!" },
    body: Buffer.from("")
  }).toString("utf8");

  expect(response).toMatchSnapshot();
});

test("builds a simple string response", () => {
  const response = buildHttpResponse({
    status: 500,
    headers: { "Content-Type": "text/plain" },
    body: Buffer.from("Internal Server Error or whatevers!")
  }).toString("utf8");

  expect(response).toMatchSnapshot()
});
