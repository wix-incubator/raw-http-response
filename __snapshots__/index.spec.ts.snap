// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`builds a simple string response 1`] = `
"HTTP/1.1 500 Internal Server Error
Content-Type: text/plain

Internal Server Error or whatevers!"
`;

exports[`builds an empty http response 1`] = `
"HTTP/1.1 200 OK
X-My-Header: Hello world!

"
`;
